src/org/zaproxy/zap/extension/ascanrules/TestCrossSiteScriptV2.java

```diff
@@ -133,6 +151,15 @@ public void init() {
         	// High level, so check all results are in the expected context
         	return hca.getHtmlContexts(attack, targetContext, ignoreFlags);
         }
+        if (findDecoded) {
+            try {
+				return hca.getHtmlContexts(URLDecoder.decode(attack, UTF8), targetContext, ignoreFlags);

```

Comment:
- Per above check, this should be done only in `HIGH`, otherwise it should use `hca.getHtmlContexts(attack)` (with attack decoded)?
 - Could use `getURLDecode(attack)` to get the decoded form.

Reply:
Still think we should do the same in `HIGH`, otherwise the double encoding will be missed in that case.